### PR TITLE
Use promises to populate panels asynchronously 

### PR DESCRIPTION
**Pull request by austinteshuba**
Closes issue #68

### CODE REVIEW

**Comment by tttzach** [comment]
LGTM - i'll be sure to do the same for the future JavaScript function calls.

File: src/main/webapp/auth.js [hunk 1]
```diff
@@ -55,16 +55,16 @@ function handleAuthenticationState() {
...
     featureContainer.removeAttribute('hidden');
 
-    // Populate information panels at top of dashboard
-    populateGmail();
-    populateTasks();
-    populateCalendar();
+    // Populate information panels at top of dashboard asynchronously
+    new Promise(() => populateGmail());
+    new Promise(() => populateTasks());
+    new Promise(() => populateCalendar());
```
Comment (gbuenoandrade) [inline src/main/webapp/auth.js:61]: Nit: I'm not super familiar with asynchronous programming in JavaScript, but it looks a bit weird to just create those promises like that...

Why have you decided not to make the `populate.*()` methods async?

Comment (antmar) [reply src/main/webapp/auth.js:61]: `populate.*()` methods are already "async" in that the calls to `fetch` return a promise and don't block.

**Comment by austinteshuba** [comment]
Closing PR as no changes are needed to achieve this - The promises already remove the blockers, and allow the main thread to continue working.

**Pull request closed without merging**